### DISCUSSION TITLE
Thoughts on the underlying pyicloud component

### DISCUSSION THREAD

**Discussion by scaraebeus** [icloud-photos-downloader/icloud_photos_downloader#735] [General]
Given the recent authentication issue #729 and dormant status on the main `picklepete/pyicloud` repository, along with some features and functionality that seem to be ahead of that project (e.g. library support, improved item types) should we look to modify the `pyicloud_ipd` component to focus on the core functionality needed by `icloudpd`?

Potential options:

- Continue to ensure all services and functionality (used or not) is fully maintained in `pyicloud_ipd`
- Trim `pyicloud_ipd` to only the services and functionality needed for the core `icloudpd`
- Find an alternative project that can replace `pyicloud_ipd` and is regularly maintained
- Leave the `pyicloud_ipd` as is and only maintain the needed services and functionality - potentially resulting in broken unused components over time

My thoughts are to trim the pyicloud_ipd component to just the needed services to handle photo downloading.  This would reduce the effort needed to maintain the component overall and create a cleaner solution.

An alternative solution would be to find another project that is maintained to offload effort and maintenance (e.g. `mandarons/icloudpy`) at the risk that it falls dormant as well.

Realistically, given the project scope, size, and resources - the answer likely sits in the "If it's not broke, don't fix it" and "Only fix it if it's broke" lanes.

**Comment by AndreyNikiforov** [comment]
@scaraebeus thanks for looking at recent icloud protocol breakages. They are really appreciated.

> should we look to modify the pyicloud_ipd component to focus on the core functionality needed by icloudpd?

Agree that we should limit our focus to the auth & photo functionality -- only what we use in `icloudpd` project. Cut away unnecessary code, leave it untouched, or something else -- most likely will be up to implementator.

For a long term solution I long wanted to untangle `icloudpd` logic from IO calls, so that contributions on bug fixes and enhancements would not be blocked by very-hard-to-maintain vcrs. That untangling is almost like a rewrite, so committing to it had been a known challenge to me. I suspect that using another project can face the same challenge -- a lot of code and tests to rewrite with the need to establish correctness.
